go/tasks/plugins/k8s/sagemaker/hpo_job.go

```diff
@@ -0,0 +1,319 @@
+	// TODO build the actual spec of the k8s resource from the taskCtx Some helpful code is already added
+	taskTemplate, err := taskCtx.TaskReader().Read(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to fetch task specification")
+	} else if taskTemplate == nil {
+		return nil, errors.Errorf("nil task specification")
+	}
+
+	taskInput, err := taskCtx.InputReader().Get(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to fetch task inputs")
+	}
+
+	// Get inputs from literals
+	inputLiterals := taskInput.GetLiterals()
+	trainPathLiteral, ok := inputLiterals["train"]
+	if !ok {
+		return nil, errors.Errorf("Input not specified: [train]")
+	}
+	validatePathLiteral, ok := inputLiterals["validation"]
+	if !ok {
+		return nil, errors.Errorf("Input not specified: [validation]")
+	}
+	staticHyperparamsLiteral, ok := inputLiterals["static_hyperparameters"]
+	if !ok {
+		return nil, errors.Errorf("Input not specified: [static_hyperparameters]")
+	}
+
+	trainingJobStoppingConditionLiteral, ok := inputLiterals["stopping_condition"]
+	if !ok {
+		return nil, errors.Errorf("Input not specified: [stopping_condition]")
+	}
+
+	hpoJobConfigLiteral, ok := inputLiterals["hpo_job_config"]
+	if !ok {
+		return nil, errors.Errorf("Input not specified: [hpo_job_config]")
+	}
+
+	outputPath := createOutputPath(taskCtx.OutputWriter().GetOutputPrefixPath().String())
+
+	// Unmarshal the custom field of the task template back into the HPOJob struct generated in flyteidl
+	sagemakerHPOJob := sagemakerSpec.HPOJob{}
+	err = utils.UnmarshalStruct(taskTemplate.GetCustom(), &sagemakerHPOJob)
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid task specification for taskType [%s]", sagemakerTaskType)
+	}
+
+	// Convert the hyperparameters to the spec value
+	staticHyperparams, err := convertStaticHyperparamsLiteralToSpecType(staticHyperparamsLiteral)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not convert static hyperparameters to spec type")
+	}
+
+	trainingJobStoppingCondition, err := convertStoppingConditionToSpecType(trainingJobStoppingConditionLiteral)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to convert stopping condition literal to spec type")
+	}
+
+	// hpo_job_config is marshaled into a byte array in flytekit, so will have to unmarshal it back
+	hpoJobConfig, err := convertHPOJobConfigToSpecType(hpoJobConfigLiteral)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to convert hpo job config literal to spec type")
+	}
+
+	// If the container is part of the task template you can access it here
+	container := taskTemplate.GetContainer()
+
+	// When adding env vars there are some default env vars that are available, you can pass them through
+	envVars := flytek8s.DecorateEnvVars(ctx, flytek8s.ToK8sEnvVar(container.GetEnv()), taskCtx.TaskExecutionMetadata().GetTaskExecutionID())
+	_ = envVars
```

Comment:
?

Reply:
This comes from the original implementation in your awsflyteplugin repo. I wasn't sure what the purpose was so I kept it here for now. Is it just for demonstration purpose?